src/main/groovy/org/openmicroscopy/tasks/IcePythonTask.groovy

```diff
@@ -1,75 +1,78 @@
-package com.zeroc.gradle.icebuilder.slice
+package org.openmicroscopy.tasks
 
+import com.zeroc.gradle.icebuilder.slice.SliceExtension
 import org.apache.commons.io.FilenameUtils
 import org.gradle.api.DefaultTask
 import org.gradle.api.GradleException
-import org.gradle.api.file.FileCollection
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.logging.Logging
+import org.gradle.api.provider.Property
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.InputFiles
 import org.gradle.api.tasks.Optional
 import org.gradle.api.tasks.OutputDirectory
 import org.gradle.api.tasks.TaskAction
 import org.gradle.api.tasks.incremental.IncrementalTaskInputs
 
-class PythonTask extends DefaultTask {
+class IcePythonTask extends DefaultTask {
 
-    private static final def Log = Logging.getLogger(PythonTask)
+    private static final def Log = Logging.getLogger(IcePythonTask)
 
     @InputFiles
-    FileCollection inputFiles
+    final ConfigurableFileCollection sourceFiles = project.files()
 
-    @OutputDirectory
-    File outputDir
-
-    @Input
+    @InputFiles
     @Optional
-    String prefix
+    final ConfigurableFileCollection includeDirs = project.files()
```

Comment:
Why 2 different approaches for python task and doc task? It will be easier to follow the same pattern for both so people can contribute easily
My pref goes to the python task way

Reply:
I agree. A base class with common options between ice and python tasks would be ideal. I would rather leave it for a separate PR however.